internal/resolution/variablesources/installed_package_test.go

```diff
@@ -3,116 +3,140 @@ package variablesources_test
+			Properties: []property.Property{
+				{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "test-package", "version": "3.0.0"}`)},
+			}},
+			InChannels: []*catalogmetadata.Channel{&channel},
+		},
+		{Bundle: declcfg.Bundle{
+			Name:    "test-package.v2.0.0",
+			Package: "test-package",
+			Image:   "registry.io/repo/test-package@v2.0.0",
+			Properties: []property.Property{
+				{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "test-package", "version": "2.0.0"}`)},
+			}},
+			InChannels: []*catalogmetadata.Channel{&channel},
+		},
+		{Bundle: declcfg.Bundle{
+			Name:    "test-package.v2.1.0",
+			Package: "test-package",
+			Image:   "registry.io/repo/test-package@v2.1.0",
+			Properties: []property.Property{
+				{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "test-package", "version": "2.1.0"}`)},
+			}},
+			InChannels: []*catalogmetadata.Channel{&channel},
+		},
+		{Bundle: declcfg.Bundle{
+			Name:    "test-package.v2.2.0",
+			Package: "test-package",
+			Image:   "registry.io/repo/test-package@v2.2.0",
+			Properties: []property.Property{
+				{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "test-package", "version": "2.2.0"}`)},
+			}},
+			InChannels: []*catalogmetadata.Channel{&channel},
+		},
+		{Bundle: declcfg.Bundle{
+			Name:    "test-package.v4.0.0",
+			Package: "test-package",
+			Image:   "registry.io/repo/test-package@v4.0.0",
+			Properties: []property.Property{
+				{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "test-package", "version": "4.0.0"}`)},
+			}},
+			InChannels: []*catalogmetadata.Channel{&channel},
+		},
+		{Bundle: declcfg.Bundle{
+			Name:    "test-package.v5.0.0",
+			Package: "test-package",
+			Image:   "registry.io/repo/test-package@v5.0.0",
+			Properties: []property.Property{
+				{Type: property.TypePackage, Value: json.RawMessage(`{"packageName": "test-package", "version": "5-0.0"}`)},
+			}},
+			InChannels: []*catalogmetadata.Channel{&channel},
+		},
+	}
+
+	const bundleImage = "registry.io/repo/test-package@v2.0.0"
+	fakeCatalogClient := testutil.NewFakeCatalogClient(bundleList)
+
+	t.Run("with ForceSemverUpgradeConstraints feature gate disabled", func(t *testing.T) {
+		defer featuregatetesting.SetFeatureGateDuringTest(t, features.OperatorControllerFeatureGate, features.ForceSemverUpgradeConstraints, false)()
+
+		ipvs, err := variablesources.NewInstalledPackageVariableSource(&fakeCatalogClient, bundleImage)
+		assert.NoError(t, err)
 
-	It("should return the correct package variable", func() {
 		variables, err := ipvs.GetVariables(context.TODO())
-		Expect(err).NotTo(HaveOccurred())
-		Expect(variables).To(HaveLen(1))
-		reqPackageVar, ok := variables[0].(*olmvariables.InstalledPackageVariable)
-		Expect(ok).To(BeTrue())
-		Expect(reqPackageVar.Identifier()).To(Equal(deppy.IdentifierFromString("installed package test-package")))
+		assert.NoError(t, err)
```

Comment:
Might need to use the `require` package rather than the `assert` package if the test is unable to continue when an assertion fails.

Reply:
I had no idea `require` package existed :) Good to know. Updated.